### ISSUE TITLE
Sync value log file before creating the next one

### ISSUE THREAD

**Issue by peterstace** [dgraph-io/badger#153]
ALICE detected a crash vulnerability between handover between value log files.

When log file `n` becomes full, we close that log file. We then create log file `n+1`. We never sync log file `n`, so it's possible for a cash state to have an unflushed (possibly empty) `n`, and a populated `n+1`. This stops badger starting.

**Comment by peterstace** [comment]
Example of strace from ALICE. Note the absence of `fsync` between operations 12 and 13.

```
0       stdout("'start:big\n'")
1       creat("LOCK", parent=1355754, mode='0666', inode=1359282)          
2       append("LOCK", offset=0, count=6, inode=1359282)
3       creat("MANIFEST", parent=1355754, mode='0666', inode=1359283)
4       trunc("MANIFEST", initial_size=0, inode=1359283, final_size=0)
5       fsync(".", size=100, inode=1355754)
6       creat("000000.vlog", parent=1355754, mode='0666', inode=1359288)
7       fsync(".", size=120, inode=1355754)
8       append("000000.vlog", offset=0, count=1048670, inode=1359288)
9       append("000000.vlog", offset=1048670, count=1048670, inode=1359288)
10      append("000000.vlog", offset=2097340, count=1048670, inode=1359288)
11      append("000000.vlog", offset=3146010, count=1048670, inode=1359288)
12      append("000000.vlog", offset=4194680, count=1048670, inode=1359288)
13      creat("000001.vlog", parent=1355754, mode='0666', inode=1359291)
14      fsync(".", size=140, inode=1355754)
15      append("000001.vlog", offset=0, count=1048670, inode=1359291)
16      creat("000001.sst", parent=1355754, mode='0666', inode=1359294)
17      fsync(".", size=160, inode=1355754)
18      append("000001.sst", offset=0, count=470, inode=1359294)
19      file_sync_range("000001.sst", count=470, offset=0, inode=1359294)
20      append("MANIFEST", offset=0, count=8, inode=1359283)
21      fsync("MANIFEST", size=8, inode=1359283)
22      unlink("LOCK", parent=1355754, inode=1359282, size=6, hardlinks=1)
23      fsync(".", size=160, inode=1355754)
24      fsync(".", size=160, inode=1355754)
25      stdout("'stop:big\n'")
```

**Comment by peterstace** [comment]
Fixed in PR #157
